config/initializers/06-pending_migrations_check.rb

```diff
@@ -0,0 +1,40 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+# We want to abort booting when there are missing migrations by default
+# since it can lead to runtime schema cache issues.
+# Refusing to boot will encourage admins to fix missing migrations.
+
+exceptions = %w(db:create db:drop db:migrate db:structure:load db:schema:load)
+applicable = exceptions.none? { |ex| ARGV.include? ex }
+
+if applicable
+  ActiveRecord::Migration.check_pending! # will raise an exception and abort boot
```

Comment:
This interferes with the normal process pending migrations are covered in rails via `config.active_record.migration_error = :page_load`

It will probably not make a real difference but I'd still prefer if we where to limit the behaviour to the prod environment.